lib/display.ex

```diff
@@ -21,9 +21,24 @@ defmodule Display do
   defp name(module), do: name(Atom.to_string(module))
 
   def show_failure(failure, module, name) do
+    intro(module, Tracker.visited)
     IO.puts(format(failure, module, name))
   end
 
+  defp intro(module, modules) do
+    if ! module in modules do
```

Comment:
Could you clarify the order of operations here? It's possible to read it as `if !(module in modules)` and `if (!module) in modules`. I don't know if this is possible, but I'd imagine you can also do `if module not in modules` or something like that.